big_tests/tests/mongoose_user_stats_SUITE.erl

```diff
@@ -82,16 +91,42 @@ user_stats_are_reported_to_google_analytics_when_mim_starts(_Config) ->
     % Restart MIM will not work, because on restart config file is read and config is reloaded,
     %  which overwrites the config and the passed option is not set
     % WHEN
-    mongoose_helper:successful_rpc(mongoose_user_stats, report, []),
+    trigger_reporting(mim()),
     % THEN
+
+    mongoose_helper:wait_until(fun no_more_events_is_reported/0, true),
     mongoose_helper:wait_until(fun hosts_count_is_reported/0, true),
     mongoose_helper:wait_until(fun modules_are_reported/0, true),
+
+    all_event_have_the_same_client_id(),
+    ok.
+
+all_clustered_mongooses_report_the_same_client_id(_Config) ->
+    trigger_reporting(mim()),
+    trigger_reporting(mim2()),
+
+    mongoose_helper:wait_until(fun no_more_events_is_reported/0, true),
+
+    all_event_have_the_same_client_id(),
     ok.
 
+
 %%--------------------------------------------------------------------
 %% Helpers
 %%--------------------------------------------------------------------
 
+all_event_have_the_same_client_id() ->
+    Tab = ets:tab2list(?ETS_TABLE),
+    UniqueSortedTab = lists:usort([Cid ||#event{cid = Cid} <- Tab]),
+    1 = length(UniqueSortedTab).
+
+no_more_events_is_reported() ->
+    Prev = get_events_collection_size(),
+    timer:sleep(100),
```

Comment:
Why do we need this sleep? Can we change it to an active waiting? I have a hunch that it will bit us on CI sooner or later.

Reply:
This actually is an attempt to make active waiting:
I wait until there is more than 0 reports than wait until I get no more reports, not receiving any more reports in 100 ms period, so I can assume all the reports were already reported. And all this is done in `mongoose_helper:wait_until/2` so each period lasts for about 150 ms.